.github/workflows/build.yml

```
@@ -22,171 +21,176 @@ env:
-        if: ${{ success() || steps.build.conclusion == 'success' }}
+        if: ${{ success() || steps.install.conclusion == 'success' }}
         run: |
-          . ../firedrake_venv/bin/activate
+          . venv/bin/activate
           firedrake-run-split-tests 5 2 "$EXTRA_PYTEST_ARGS --junit-xml=firedrake5_{#}.xml"
+        timeout-minutes: 15
 
       - name: Run tests (nprocs = 6)
-        if: ${{ success() || steps.build.conclusion == 'success' }}
+        if: ${{ success() || steps.install.conclusion == 'success' }}
         run: |
-          . ../firedrake_venv/bin/activate
+          . venv/bin/activate
           firedrake-run-split-tests 6 2 "$EXTRA_PYTEST_ARGS --junit-xml=firedrake6_{#}.xml"
+        timeout-minutes: 15
 
       - name: Run tests (nprocs = 7)
-        if: ${{ success() || steps.build.conclusion == 'success' }}
+        if: ${{ success() || steps.install.conclusion == 'success' }}
         run: |
-          . ../firedrake_venv/bin/activate
+          . venv/bin/activate
           firedrake-run-split-tests 7 1 "$EXTRA_PYTEST_ARGS --junit-xml=firedrake7_{#}.xml"
+        timeout-minutes: 15
 
       - name: Run tests (nprocs = 8)
-        if: ${{ success() || steps.build.conclusion == 'success' }}
+        if: ${{ success() || steps.install.conclusion == 'success' }}
         run: |
-          . ../firedrake_venv/bin/activate
+          . venv/bin/activate
           firedrake-run-split-tests 8 1 "$EXTRA_PYTEST_ARGS --junit-xml=firedrake8_{#}.xml"
+        timeout-minutes: 15
 
       - name: Run Gusto smoke tests
         # Only test Gusto in real mode
-        if: (success() || steps.build.conclusion == 'success') && matrix.scalar-type == 'real'
-        run: |
-          . ../firedrake_venv/bin/activate
-          cd ../firedrake_venv/src/gusto
-          python -m pytest -n 12 -v \
-            integration-tests/balance/test_saturated_balance.py \
-            integration-tests/equations/test_thermal_sw.py \
-            integration-tests/transport/test_embedded_dg_advection.py
+        if: (success() || steps.install.conclusion == 'success') && matrix.arch == 'default'
+        run: |
+          . venv/bin/activate
+          git clone --depth 1 https://github.com/firedrakeproject/gusto.git gusto-repo
+          pip install --verbose ./gusto-repo
+          python -m pytest -n 12 --verbose \
+            gusto-repo/integration-tests/balance/test_saturated_balance.py \
+            gusto-repo/integration-tests/equations/test_thermal_sw.py \
+            gusto-repo/integration-tests/transport/test_embedded_dg_advection.py
+        timeout-minutes: 10
+      
+      - name: Run spyro smoke tests
+        if: ${{ success() || steps.build.conclusion == 'success' }}
+        run: |
+          . venv/bin/activate
+          git clone --depth 1 https://github.com/NDF-Poli-USP/spyro.git spyro-repo
+          pip install --verbose ./spyro-repo
+          mpiexec -n 6 python -m pytest  spyro-repo/test_integration/
```

```suggestion
        if: success() || steps.install.conclusion == 'success'
        run: |
          . venv/bin/activate
          git clone --depth 1 https://github.com/NDF-Poli-USP/spyro.git spyro-repo
          pip install --verbose ./spyro-repo
          mpiexec -n 6 python -m pytest spyro-repo/test_integration/
```

Note that this will test in both real and complex mode. Is that desired?

---

Not really, so I will change it. Thanks!